ratis-server/src/main/java/org/apache/ratis/server/impl/RaftServerImpl.java

```diff
@@ -1805,17 +1805,17 @@ TransactionContext getTransactionContext(LogEntryProto entry, Boolean createNew)
   CompletableFuture<Message> applyLogToStateMachine(ReferenceCountedObject<LogEntryProto> nextRef)
       throws RaftLogIOException {
     LogEntryProto next = nextRef.get();
-    if (!next.hasStateMachineLogEntry()) {
-      stateMachine.event().notifyTermIndexUpdated(next.getTerm(), next.getIndex());
-    }
-
-    if (next.hasConfigurationEntry()) {
+    switch (next.getLogEntryBodyCase()) {
```

Comment:
It is good to use `switch`.  Let add also `case METADATAENTRY` and make `default` throw an exception.